From: Marc Zyngier <marc.zyngier@arm.com>
Date: Mon, 13 Jun 2016 10:39:44 +0100
Subject: genirq: Look-up percpu trigger type if not specified by caller
Git-commit: f35ad083783e8ed6ac030f5feb209f864875b413
Patch-mainline: v4.8-rc1
References: fate#322150

As we now do for non-percpu interrupt, perform a lookup of the
interrupt trigger if the user doesn't supply one. The difference
here is that we can only do it at enable time (trigger configuration
can be per-cpu as well).

Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 kernel/irq/manage.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index eaedeb7..f78b084 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -1737,7 +1737,14 @@ void enable_percpu_irq(unsigned int irq, unsigned int type)
 	if (!desc)
 		return;
 
+	/*
+	 * If the trigger type is not specified by the caller, then
+	 * use the default for this interrupt.
+	 */
 	type &= IRQ_TYPE_SENSE_MASK;
+	if (type == IRQ_TYPE_NONE)
+		type = irqd_get_trigger_type(&desc->irq_data);
+
 	if (type != IRQ_TYPE_NONE) {
 		int ret;
 
-- 
2.10.0

